### PR TITLE
Implement an end-to-end zkFold Symbolic smart contract test on testnet.

### PR DESCRIPTION
**Pull request by zlonast**

### CODE REVIEW

File: src/ZkFold/Cardano/ScriptsVerifier.hs [hunk 1]
```diff
@@ -1,62 +1,87 @@
...
+makeLift ''RedeemerVerifier
+makeIsDataIndexed ''RedeemerVerifier [('RedeemerVerifier, 0)]
 
 -- TODO: split the setup data into the fixed and varying parts
 -- | The Plutus script for verifying a ZkFold Symbolic smart contract.
 {-# INLINABLE symbolicVerifier #-}
-symbolicVerifier :: Setup PlonkPlutus -> Input PlonkPlutus -> Proof PlonkPlutus -> ScriptContext -> Bool
-symbolicVerifier contract input proof ctx = condition1 && condition2
+symbolicVerifier :: DatumVerifier -> RedeemerVerifier -> ScriptContext -> Bool
+symbolicVerifier _ (RedeemerVerifier contract input proof) ctx =
```
Comment (vlasin) [inline src/ZkFold/Cardano/ScriptsVerifier.hs:44]: Why do we need datum here?

File: src/ZkFold/Cardano/ScriptsVerifier.hs [hunk 2]
```diff
@@ -1,62 +1,87 @@
...
         -- Verifying the validity of the ZkFold Symbolic smart contract on the current transaction.
         -- The smart contract is encoded into the `Setup PlonkPlutus` data structure.
         condition2 = verify @PlonkPlutus contract input proof
 
 -- | The Plutus script (minting policy) for verifying a Plonk proof.
 {-# INLINABLE plonkVerifier #-}
-plonkVerifier :: Setup PlonkPlutus -> Input PlonkPlutus -> Proof PlonkPlutus -> ScriptContext -> Bool
-plonkVerifier computation input proof ctx = condition0 && (condition1 || condition2)
+plonkVerifier :: DatumVerifier -> RedeemerVerifier -> ScriptContext -> Bool
+plonkVerifier _ (RedeemerVerifier computation input proof) ctx =
```
Comment (vlasin) [inline src/ZkFold/Cardano/ScriptsVerifier.hs:68]: Same here: datum is not used

File: src/ZkFold/Cardano/ScriptsVerifier.hs [hunk 3]
```diff
@@ -1,62 +1,87 @@
...
         [(TokenName t, n)] = AssocMap.toList m
 
         -- With this minting policy, we can mint tokens if the Plonk proof is valid for the input provided in the redeemer.
         -- The tokens serve as proof that the network has verified the computation.
         -- We can also burn already minted tokens.
 
-        -- Verifying that the token name equals to the bytestring representation of the public input in the ZKP protocol
-        condition0 = t == toTranscript (Plutus.head $ pubInput input)
+        -- Verifying that the token name equals to the bytestring representation of the public input in the ZKP protocol.
+        condition0 = t == serialiseData (toBuiltinData input)
```
Comment (vlasin) [inline src/ZkFold/Cardano/ScriptsVerifier.hs:81]: Is it faster than `bytestringToInteger`?

File: bench-cpu-mem/Bench/Scripts.hs [hunk 4]
```diff
@@ -1,32 +1,36 @@
...
 
 import           ZkFold.Base.Protocol.NonInteractiveProof (NonInteractiveProof (..))
 import           ZkFold.Cardano.Plonk                     (PlonkPlutus)
-import           ZkFold.Cardano.ScriptsVerifier           (plonkVerifier, symbolicVerifier)
+import           ZkFold.Cardano.ScriptsVerifier
 
-symbolicVerifierScript :: Setup PlonkPlutus -> Input PlonkPlutus -> Proof PlonkPlutus -> UPLC.Program UPLC.NamedDeBruijn DefaultUni DefaultFun ()
-symbolicVerifierScript s i p =
+symbolicVerifierScript :: DatumVerifier -> RedeemerVerifier -> ScriptContext -> UPLC.Program UPLC.NamedDeBruijn DefaultUni DefaultFun ()
+symbolicVerifierScript datum redeemer ctx =
```
Comment (vlasin) [inline bench-cpu-mem/Bench/Scripts.hs:19]: Do we need a datum here?

File: bench-cpu-mem/Bench/Scripts.hs [hunk 5]
```diff
@@ -1,32 +1,36 @@
...
-       `unsafeApplyCode` liftCodeDef i
-       `unsafeApplyCode` liftCodeDef p
+       `unsafeApplyCode` liftCodeDef datum
+       `unsafeApplyCode` liftCodeDef redeemer
+       `unsafeApplyCode` liftCodeDef ctx
 
-plonkVerifierScript :: Setup PlonkPlutus -> Input PlonkPlutus -> Proof PlonkPlutus -> UPLC.Program UPLC.NamedDeBruijn DefaultUni DefaultFun ()
-plonkVerifierScript s i p =
+plonkVerifierScript :: DatumVerifier -> RedeemerVerifier -> ScriptContext -> UPLC.Program UPLC.NamedDeBruijn DefaultUni DefaultFun ()
+plonkVerifierScript datum redeemer ctx =
```
Comment (vlasin) [inline bench-cpu-mem/Bench/Scripts.hs:26]: Do we need a datum here?

File: src/ZkFold/Cardano/ScriptsVerifier.hs (hunk 3 above)
Comment (zlonast) [reply src/ZkFold/Cardano/ScriptsVerifier.hs]: byteStringToInteger is faster

**Review by vlasin — APPROVED** [review]

**Pull request merged**